src/dotnet/CarbonAware.LocationSources.Azure/test/AzureLocationSourceTest.cs

```diff
@@ -46,39 +57,41 @@ public void TestToGeopositionInvalidLocation()
     /// returns original Location.
     /// </summary>
     [Test]
-    public async Task TestToGeopositionWhenAlreadyGeopositionLocation()
+    public void TestToGeopositionWhenAlreadyGeopositionLocation()
     {
         var mockLocationSource = SetupMockLocationSource().Object;
-        var result = await mockLocationSource.ToGeopositionLocation(Constants.EastUsGeoLocation);
-        Assert.AreEqual(Constants.EastUsGeoLocation, result);
+        Location location = new Location {
+            LocationType = LocationType.Geoposition
+        };
+        var result = mockLocationSource.ToGeopositionLocation(location);
+        Assert.AreEqual(location, result);
     }
 
     private static Mock<AzureLocationSource> SetupMockLocationSource() {
         var logger = Mock.Of<ILogger<AzureLocationSource>>();
         var mockLocationSource = new Mock<AzureLocationSource>(logger);
         
         mockLocationSource.Protected()
-            .Setup<Dictionary<string, AzureRegion>>("GetAzureRegions")
-            .Returns(GetTestAzureRegions())
+            .Setup<Dictionary<string, NamedGeoposition>>("LoadRegionsFromJson")
+            .Returns(GetTestDataRegions())
             .Verifiable();
 
         return mockLocationSource;
     }
 
-    private static Dictionary<string, AzureRegion> GetTestAzureRegions() {
+    private static Dictionary<string, NamedGeoposition> GetTestDataRegions() {
         // All the tests above correspond to values in this mock data. If the mock values are changed, the tests need to be updated 
-        return new Dictionary<string, AzureRegion>() {
+        return new Dictionary<string, NamedGeoposition>() {
             {"eastus", Constants.EastUsRegion },
             {"westus", Constants.WestUsRegion },
             {"northcentralus", Constants.NorthCentralRegion }
         };
     }
 
-    private static void AssertLocationsEqual(Location location1, Location location2)
+    private static void AssertLocationsEqual(NamedGeoposition data, Location result)
```

Comment:
This is confusing because `ToGeopositionLocation` takes in a `Location` and returns a `Location` so that's what is expected to be asserted on (since it's expected vs. result)